aw-transform/src/flood.rs

```diff
@@ -36,8 +36,15 @@ pub fn flood(events: Vec<Event>, pulsetime: chrono::Duration) -> Vec<Event> {
                     warn!("Gap was of negative duration ({}s), but could be safely merged. This error will only show once per batch.", gap);
                     warned_negative_gap_safe = true;
                 }
-                // Extend e1 to the middle between e1 and e2
-                e1.duration = e2.calculate_endtime() - e1.timestamp;
+                // Choose the longest event and set the endtime to it
+                // TODO: Also possibly extend to an e3 if that exists?
```

Comment:
Hmm, I thought the Python version would do this too, but maybe it doesn't?

Reply:
I'm not sure if I'm reading the code correctly, but this is how it works I think.

It iterates over the events with zip which creates an array in the form of `[[1,2],[2,3],[3,4]]`, but if it sees a change to merge an event it extends the first event to the end of the second one and sets the duration of the second one to 0 and afterwards filter out all events with an duration of 0. So that means that today it "kind of" works because it will not merge the same event over and over but create multiple merged events even if they have the same data and could be merged, but it would be easy to fix by simply extending the second event instead of the first one so on the next iteration it would have an event which does not have a duration of 0.

Reply:
In the latest commit I have fixed it.